pkgs/_macro_host/lib/macro_host.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:_macro_builder/macro_builder.dart';
+import 'package:_macro_runner/macro_runner.dart';
+import 'package:_macro_server/macro_server.dart';
+import 'package:macro_service/macro_service.dart';
+
+class MacroHost {
+  final MacroServer macroServer;
+  final MacroBuilder macroBuilder = MacroBuilder();
+  final MacroRunner macroRunner = MacroRunner();
+  late final HostEndpoint hostEndpoint;
+
+  MacroHost({required MacroHostService service})
+      : macroServer = MacroServer(service: service);
+
+  Future<void> serve() async {
```

Comment:
Consider an async static method to get an instance, and a private constructor, so that we don't need a late field and this extra function that needs to be called before the class is actually valid.

Reply:
Done, thanks.